src/t_hash.c

```diff
@@ -671,18 +601,15 @@ static int _hashListpackEntryValidation(unsigned char *p, void *userdata) {
     return 1;
 }
 
-/* Validate the integrity of the data structure.
- * when `deep` is 0, only the integrity of the header is validated.
- * when `deep` is 1, we scan all the entries one by one. */
-int hashZiplistValidateIntegrity(unsigned char *zl, size_t size, int deep) {
-    if (!deep)
-        return ziplistValidateIntegrity(zl, size, 0, NULL, NULL);
-
+/* Validate the integrity of the data structure. The function will convert
+ * ziplist to listpack and store it in 'lp' at the same time. */
+int hashZiplistValidateIntegrity(unsigned char *zl, size_t size, unsigned char **lp) {
```

Comment:
i see you did update the comment, but we also need to update the function name, and mention what it expects of the `lp` argument.

i think i would call it `hashZiplistConvertAndValidateIntegrity`

Reply:
Fixed.